jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -41,29 +40,28 @@ class FaissSearcher(Executor):
         import numpy as np
         from jina.executors.indexers.vector.faiss import FaissIndexer
 
-        train_filepath = 'faiss_train.tgz'
-        train_data = np.random.rand(10000, 128)
-        with gzip.open(train_filepath, 'wb', compresslevel=1) as f:
-            f.write(train_data.astype('float32'))
-        indexer = FaissIndexer('PCA64,FLAT', train_filepath)
+        import faiss
+        trained_index_file = os.path.join(os.environ['TEST_WORKSPACE'], 'faiss.index')
+        train_data = np.array(np.random.random([1024, 10]), dtype=np.float32)
+        faiss_index = faiss.index_factory(10, 'IVF10,PQ2')
+        faiss_index.train(train_data)
+        faiss.write_index(faiss_index, trained_index_file)
+
+        searcher = FaissSearcher('PCA64,FLAT', trained_index_file=trained_index_file)
 
-        # generate a training file in `.npy`
-        train_filepath = 'faiss_train'
-        np.save(train_filepath, train_data)
-        indexer = FaissIndexer('PCA64,FLAT', train_filepath)
     """
 
     def __init__(
         self,
         index_key: str = 'Flat',
-        train_filepath: Optional[str] = None,
```

Comment:
so we do not support anymore training at initialization right? it seems a good change to me

Reply:
yes, I created a separate training request endpoint to handle the training procedure.